frontend/src/components/common/Button.vue

```diff
@@ -1,16 +1,24 @@
-<script setup>
+<script setup lang="ts">
 import Button from 'primevue/button';
+import type { PropType } from 'vue';
+import { IconPosition } from '../../enum/IconEnum';
 const props = defineProps({
     label: {
         type: String,
         default: 'Click',
     },
+    iconPos: {
+        type: String as PropType<IconPosition>,
+        default: IconPosition.right,
+    },
 });
 </script>
 <template>
-    <Button class="nr-button"
-        ><span>{{ props.label }}</span> <slot class="icon"></slot
-    ></Button>
+    <Button class="nr-button">
+        <slot class="icon" v-if="iconPos === IconPosition.left"></slot>
+        <span>{{ props.label }}</span>
+        <slot class="icon" v-if="iconPos === IconPosition.right"></slot>
```

Comment:
Can "v-else" work?

Reply:
I tried to use a v-else but it gave the " v-else/v-else-if has no adjacent v-if or v-else-if " typescript com·plaint.

Reply:
Hmm... a bit surprised vue does not have this support.
We can only leave it as v-if then.